src/ray/raylet/node_manager.cc

```diff
@@ -831,6 +831,7 @@ void NodeManager::ProcessDisconnectClientMessage(
         // not safe to pass in the iterator directly.
         const TaskID task_id = *worker->GetBlockedTaskIds().begin();
         HandleTaskUnblocked(client, task_id);
+        RAY_CHECK(task_dependency_manager_.UnsubscribeAllDependencies(task_id));
```

Comment:
Can you add a `TODO` here that this will only unsubscribe for the currently executing task, and that we should also unsubscribe for previous tasks that still have an active `ray.wait`?